src/figuresClasses.ts

```diff
@@ -1,19 +1,99 @@
-export interface Figure {
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
 
+enum Color {
+  Red = 'red',
+  Green = 'green',
+  Blue = 'blue',
 }
 
-export class Triangle {
+export interface Figure {
+  shape: Shape,
+  color: Color,
+  getArea(): number,
+}
 
+function checkSides(...args: number[]): void {
+  if (Math.min(...args) <= 0) {
+    throw new Error('The length cannot to be negative number');
+  }
 }
 
-export class Circle {
+export class Triangle implements Figure {
+  readonly shape = Shape.Triangle;
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    checkSides(a, b, c);
+
+    const [
+      sideA,
+      sideB,
+      sideC,
+    ]: number[] = [a, b, c].sort((prev, next) => prev - next);
+
+    if (sideC >= sideB + sideA) {
+      throw new Error('Impossible to build triangle with these sides!');
+    }
+  }
 
+  getArea(): number {
+    const semiperimeter: number = (this.a + this.b + this.c) / 2;
+    const area: number = Math.sqrt(
+      semiperimeter
+      * (semiperimeter - this.a)
+      * (semiperimeter - this.b)
+      * (semiperimeter - this.c),
+    );
+
+    return +(area.toFixed(2));
```

Comment:
IMO it's a strange approach, you've used different rounding logic to get same result (here is `toFixed`, it's division by 100 in next class)

Better to create helper function for that purpose and reuse it in all cases